common/samd21g18a_firmware_partition.ld

```diff
@@ -50,7 +50,7 @@ SEARCH_DIR(.)
 MEMORY
 {
   rom    (rx)  : ORIGIN = 0x00001000, LENGTH = 0x0003efff
-  ram    (rwx) : ORIGIN = 0x20000000, LENGTH = 0x00001000
+  ram    (rwx) : ORIGIN = 0x20000000, LENGTH = 0x00008000
```

Comment:
I should have done this as a separate commit -- this just enables the use of the 32KB (0x8000 bytes) of RAM on the SAMD21G18 instead of the 4KB (0x1000 bytes) on the SAMD21G15 we originally developed for. Overlapping the flash and port buffers was a dirty hack used to save RAM when we needed it, but the pointer cast didn't guarantee alignment.